cmd/global-query/pkg/distributed/query.go

```diff
@@ -20,6 +20,8 @@ import (
 type QueryRunner struct {
 	resolver hosts.Resolver
 	querier  Querier
+
+	onResult func(*results.Result) error
```

Comment:
Since this is a crucial element (basically adding event-driven handling to our proceedings) maybe it would make sense to add a short comment even though it's not exported?